yt_dlp/extractor/noodlemagazine.py

```diff
@@ -37,15 +38,21 @@ def _real_extract(self, url):
         like_count = parse_count(self._html_search_meta('ya:ovs:likes', webpage, default=None))
         upload_date = unified_strdate(self._html_search_meta('ya:ovs:upload_date', webpage, default=''))
 
-        key = self._html_search_regex(rf'/{video_id}\?(?:.*&)?m=([^&"\'\s,]+)', webpage, 'key')
-        playlist_info = self._download_json(f'https://adult.noodlemagazine.com/playlist/{video_id}?m={key}', video_id)
+        player_url = self._html_search_regex(rf'(/player/{video_id}[^"\'\s,]+)', webpage, 'player-url')
+        player_iframe = self._download_webpage(f'https://adult.noodlemagazine.com{player_url}', video_id)
+        playlist_url = self._html_search_regex(rf'(/playlist/{video_id}[^"\'\s,]+)', player_iframe, 'playlist-url')
+        playlist_info = self._download_json(f'https://adult.noodlemagazine.com{playlist_url}', video_id)
```

Comment:
we can add the `Referer` header to this API call to prevent the 403
```suggestion
        playlist_info = self._download_json(
            urljoin('https://adult.noodlemagazine.com', playlist_url), video_id, headers={'Referer': url})
```